### PR TITLE
Cht tests fazzdev #1321 41

### PR DESCRIPTION
**Pull request by thefazzer**
**Cht tests fazzdev #1321 41**

- BlockDataDownloaded called on 10 known blocks.
- Make sure that UnconsumedBlocksDataBytes is equal to the sum of serialized sizes of those blocks.

Preferred: @Aprogiena @noescape00

### CODE REVIEW

File: src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs [hunk 1]
```diff
@@ -1652,5 +1652,38 @@ public void ChainWith10DataAvailableHeaders_PartialValidationSucceededOnTheFirst
...
+
+            // Chain header tree setup.
+            TestContext testContext = new TestContextBuilder().WithInitialChain(initialChainSize).UseCheckpoints(false).Build();
+            ChainedHeaderTree chainedHeaderTree = testContext.ChainedHeaderTree;
+            ChainedHeader initialChainTip = testContext.InitialChainTip;
+
+            ChainedHeader extendedChainTip = testContext.ExtendAChain(chainExtension, initialChainTip);
+            List<BlockHeader> listOfChainBlockHeaders = testContext.ChainedHeaderToList(extendedChainTip, initialChainSize + chainExtension);
+
+            // Ten headers beyond the initial chain presented h6->h15.
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs]: The comment does not match the code as you are presenting the whole chain.

Comment (thefazzer) [reply src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs]: Agreed - fixed.

**Comment by Aprogiena** [comment]
Good now.

**Comment by thefazzer** [comment]
Merge conflicts fixed.

**Pull request merged**